Add timers to the TestResult, and update the TRX logger to output the timer information

robdalsanto opened the pull request:
Add timers to the TestResult, and update the TRX logger output the timer 
information to the TRX file if any timers are in the collection, as per 
the XML schema of the TRX from
%VSINSTALLDIR%\xml\Schemas\vstst.xsd

It looks almost like an oversight that the TestResult did not previously include
the named timers.

## Related issues
- The primary motivation for this connects back to [testfx#504](https://github.com/microsoft/testfx/issues/504)
- To support testfx#504, the TRX logger has to be updated, as per #2196.

Commit:
Add timers to the TestResult, and update the TRX logger to
output the timer information to the TRX file if any timers
are in the collection, as per the XML schema of the TRX from
%VSINSTALLDIR%\xml\Schemas\vstst.xsd

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/microsoft/vstest?pullRequest=2208) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/vstest?pullRequest=2208) before we can accept your contribution.<br/><br/>:x: robdalsanto [sign now](https://cla.opensource.microsoft.com/microsoft/vstest?pullRequest=2208)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/microsoft/vstest?pullRequest=2208) it.</sub>

Commit:
Previous commit missed change to expected result in TestResult serialization
unit test for added property.

robdalsanto commented:
Added fix for failing unit test by adding commit 'df5213197b38378a712a544a6d095975cd737e31' to the pull request.

singhsarab commented on src/Microsoft.TestPlatform.ObjectModel/Resources/Resources.resx:
```diff
@@ -286,6 +286,10 @@
     <value>  {0}:
 {1}</value>
   </data>
+  <data name="TestResultTimerFormat" xml:space="preserve">
```
This does have to be a resources.
Resources are strings that need to be localized. We should simply use a const for this.

robdalsanto replied:
Understood and I agree. I'll make that update.

robdalsanto replied:
Done in commit 1cbbb23d5b40eebcf18cecfdfcb89f32680ba93f

singhsarab commented on src/Microsoft.TestPlatform.ObjectModel/TestResult.cs:
```diff
@@ -309,6 +332,26 @@ public string Text
         }
     }
 
+    [DataContract]
```
Docs/Comments

robdalsanto replied:
Yes, added them.

Commit:
Address code review comments from @singhsarab.
Removed a RESX string and replaced with a literal string, and
added summary comments.

robdalsanto commented:
@singhsarab I addressed your review comments. Thanks for the input. 
It looks like there was some problem in the build unrelated to my changes (see [here](https://dev.azure.com/vstestplatform/TestPlatform/_build/results?buildId=860&view=logs&jobId=2d2b3007-3c5c-5840-9bb0-2b1ea49925f3&taskId=7376a427-0fb6-5027-9e88-fb5bb628b2d5&lineStart=277&lineEnd=301&colStart=1&colEnd=43)). I can't find a way to re-queue a build. Can you provide me any guidance so this doesn't stall ?

singhsarab commented:
@robdalsanto Could you please attach a sample trx with the change ?

singhsarab commented:
We need some tests, acceptance and unit for this change.

HeroMaxPower commented:
@robdalsanto could you make some acceptance tests?

robdalsanto commented:
Yes, I am busy now until June. So while I plan to get to this, it won't be
right away.

On Mon, Mar 9, 2020 at 11:29 AM Maximilian Chaplin <notifications@github.com>
wrote:

> @robdalsanto <https://github.com/robdalsanto> could you make some
> acceptance tests?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/microsoft/vstest/pull/2208?email_source=notifications&email_token=AHOFEGC7XYWMSMWWI4MWKZDRGUYOXA5CNFSM4I4ORNFKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEOIPMWQ#issuecomment-596702810>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHOFEGESXN6VWJGKESSNZZTRGUYOXANCNFSM4I4ORNFA>
> .
>

nohwnd commented:
Looks like a nice change, that is only missing tests, adding for consideration in 16.7.0.

Evangelink commented:
@robdalsanto We know it's been a really long time but would you be able to work on the tests or do you want us to take it over?

robdalsanto commented:
Hi,
Sorry to punt, but yes please take over and create the tests. I thought I'd
have time to write them, and agree they are needed. But I don't see an
opportunity to fit the task into my schedule anytime soon.
Thanks,
Rob

On Wed, Aug 3, 2022, 11:58 AM Amaury Levé ***@***.***> wrote:

> @robdalsanto <https://github.com/robdalsanto> We know it's been a really
> long time but would you be able to work on the tests or do you want us to
> take it over?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/microsoft/vstest/pull/2208#issuecomment-1204063265>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHOFEGH64VIPJLAACKX7MFTVXKCKXANCNFSM4I4ORNFA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

Evangelink commented:
@robdalsanto No worries, thanks for letting us know!

MarcoRossignoli commented:
I'm going to close for stale state.
